src/nemo_evaluator/reports/trajectories.py

```
@@ -196,6 +423,55 @@ def _index_traffic_by_trial(
+    traffic_by_trial_all: dict[tuple[Any, Any], list[dict[str, Any]]],
+    selected_sessions_by_trial: dict[tuple[Any, Any], str],
+    *,
+    only_successful: bool = True,
+) -> dict[tuple[Any, Any], list[dict[str, Any]]]:
+    selected_rows: dict[tuple[Any, Any], list[dict[str, Any]]] = {}
+    for key, rows in traffic_by_trial_all.items():
+        if key not in selected_sessions_by_trial:
+            continue
+        selected_session_id = selected_sessions_by_trial[key]
+        if selected_session_id:
+            rows = [row for row in rows if _wire_session_id(row) == selected_session_id]
```

Ruff (0.15.18)

[warning] `for` loop variable `rows` overwritten by assignment target

(PLW2901)